### PR TITLE
feat: Added draw landscape option

### LINKED ISSUE #462: Draw Landscape Boundary: Add Draw landscape boundary map option
## Description
- Add new option to draw landscape boundary

## Additional context

Figma: https://www.figma.com/file/BYn1r2xSKtap9V6pefEMiz/Draw-landscape---R1.3?node-id=502%3A128
POC: https://github.com/techmatters/terraso-web-client/compare/main...poc/polygon-user-test?expand=1

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Added Draw landscape polygon option

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [x] English strings reviewed and copyedited
- [x] Strings are localized
- [x] Verified on mobile
- [x] Verified on desktop
- [x] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #462 

### Verification steps

Screen Shot 2022-08-02 at 11 54 55 AM
Screen Shot 2022-08-02 at 11 54 47 AM
Screen Shot 2022-08-02 at 11 54 29 AM

### CODE REVIEW

File: src/landscape/components/LandscapeListMap.css [hunk 1]
```diff
@@ -11,15 +11,15 @@
...
-  background-color: #98532999;
+  background: #98532999;
   border-radius: 20px;
 }
 
-.landscape-list-map-marker-icon {
+.landscape-list-map-marker-icon.leaflet-marker-icon.leaflet-interactive {
   background-clip: padding-box;
-  background-color: #985329;
+  background: #985329;
```
Comment (paulschreiber) [inline src/landscape/components/LandscapeListMap.css:22]: question: did you mean to change this? You're also overriding repeat, position, etc.

File: src/localization/locales/en-US.json [hunk 2]
```diff
@@ -395,7 +401,116 @@
...
+                    },
+                    "marker": {
+                        "tooltip": {
+                            "start": "Click map to place marker."
+                        }
+                    },
+                    "polygon": {
+                        "tooltip": {
+                            "start": "Click to start drawing shape.",
+                            "cont": "Click to continue drawing shape.",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:452]:
```suggestion
                            "start": "Start drawing shape.",
                            "cont": "Continue drawing shape.",
```

File: src/localization/locales/en-US.json [hunk 3]
```diff
@@ -395,7 +401,116 @@
...
+                        "title": "Delete last point drawn",
+                        "text": "Delete last point"
+                    },
+                    "buttons": {
+                        "polyline": "Draw a polyline",
+                        "polygon": "Draw a polygon",
+                        "rectangle": "Draw a rectangle",
+                        "circle": "Draw a circle",
+                        "marker": "Draw a marker",
+                        "circlemarker": "Draw a circlemarker"
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:429]: issue: remove these, as we aren't using them.

File: src/localization/locales/en-US.json [hunk 4]
```diff
@@ -395,7 +401,116 @@
...
+                    "finish": {
+                        "title": "Finish drawing",
+                        "text": "Close Shape"
+                    },
+                    "undo": {
+                        "title": "Delete last point drawn",
+                        "text": "Delete last point"
+                    },
+                    "buttons": {
+                        "polyline": "Draw a polyline",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:424]: issue: remove this, as we aren't using it.

File: src/localization/locales/en-US.json [hunk 5]
```diff
@@ -395,7 +401,116 @@
...
+                    "circlemarker": {
+                        "tooltip": {
+                            "start": "Click map to place circle marker."
+                        }
+                    },
+                    "marker": {
+                        "tooltip": {
+                            "start": "Click map to place marker."
+                        }
+                    },
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:448]: issue: remove these, as we aren't using them.

File: src/localization/locales/en-US.json [hunk 6]
```diff
@@ -395,7 +401,116 @@
...
+                    "rectangle": {
+                        "tooltip": {
+                            "start": "Click and drag to draw rectangle."
+                        }
+                    },
+                    "simpleshape": {
+                        "tooltip": {
+                            "end": "Release mouse to finish drawing."
+                        }
+                    }
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:473]: issue: remove these, as we aren't using them.

File: src/localization/locales/en-US.json [hunk 7]
```diff
@@ -293,7 +293,14 @@
         "list_map_help_url": "https://terraso.org/help/add-a-new-landscape-to-terraso/",
         "list_map_section_label": "Landscapes map",
         "list_map_popup_link": "View details about {{name}}",
-        "boundaries_update_save": "Update Map"
+        "boundaries_update_save": "Update Map",
+        "form_boundary_draw_polygon_title": "[TODO] Draw {{name}}’s Boundary",
+        "form_boundary_draw_polygon_description": "[TODO] Click anywhere to start drawing. Save your drawing and return to complete later. You can adjust the boundary in edit mode. ",
+        "form_boundary_draw_polygon_help": "[TODO] Need help?",
+        "form_boundary_draw_polygon_help_url": "https://terraso.org/help/",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:300]: question: Should this go to drawing-specific help?

File: src/localization/locales/en-US.json [hunk 8]
```diff
@@ -293,7 +293,14 @@
         "list_map_help_url": "https://terraso.org/help/add-a-new-landscape-to-terraso/",
         "list_map_section_label": "Landscapes map",
         "list_map_popup_link": "View details about {{name}}",
-        "boundaries_update_save": "Update Map"
+        "boundaries_update_save": "Update Map",
+        "form_boundary_draw_polygon_title": "[TODO] Draw {{name}}’s Boundary",
+        "form_boundary_draw_polygon_description": "[TODO] Click anywhere to start drawing. Save your drawing and return to complete later. You can adjust the boundary in edit mode. ",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:298]: issue: this is not accurate. You can't click anywhere. You need to click the polygon tool first. It's also ungrammatical. Needs complete rewrite.

File: src/localization/locales/en-US.json [hunk 9]
```diff
@@ -293,7 +293,14 @@
...
+        "form_boundary_draw_polygon_help": "[TODO] Need help?",
+        "form_boundary_draw_polygon_help_url": "https://terraso.org/help/",
+        "form_boundary_options_draw_polygon": "Draw the landscape’s boundary on a map",
+        "form_boundary_draw_polygon_updated": "Landscape boundary is complete. Proceed to update your landscape's map or edit the shape.",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:302]: issue: second sentence is confusing. Remove or rewrite.

File: src/localization/locales/en-US.json [hunk 10]
```diff
@@ -293,7 +293,14 @@
...
+        "form_boundary_draw_polygon_edit_help": "<strong>How to edit:</strong> Drag a translucent point to add more points. Click the point to remove it. When you’re done adjusting the boundary save to exit the edit mode."
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:303]: This is confusing. Users won't understand the concept of "edit mode."

**Review by paulschreiber — CHANGES_REQUESTED** [review]
Other problems:
* flyouts over polygon tool persists during drawing, blocking the shape
* clicking clear all and then clicking the polygon tool and the deleted shape reappears
* permits creation of multiple polygons

File: src/landscape/components/LandscapeListMap.css (hunk 1 above)
Comment (josebui) [reply src/landscape/components/LandscapeListMap.css:22]: @paulschreiber yes, this was intentional to override a conflicting style in the Map.css file, should I override something else?

File: src/localization/locales/en-US.json (hunk 2 above)
Comment (josebui) [reply src/localization/locales/en-US.json:452]: @paulschreiber I am not sure if this is better, I think the message intention is to make it clear that they have to click, without that action seems like you are already perfomring the action, can you add more context on why you think this is better?

File: src/localization/locales/en-US.json (hunk 7 above)
Comment (josebui) [reply src/localization/locales/en-US.json:300]: @paulschreiber Hana asked to put that general link, see the converastion here: https://tech-matters.slack.com/archives/C02K02L3Q48/p1659022864655649

**Comment by josebui** [comment]
@paulschreiber About your other problems comment:

`flyouts over polygon tool persists during drawing, blocking the shape:` Is this about the popup when you create or edit a polygon?

`clicking clear all and then clicking the polygon tool and the deleted shape reappears`: I fixed this

`permits creation of multiple polygons`: This is the expected behavior

File: src/landscape/components/LandscapeListMap.css (hunk 1 above)
Comment (paulschreiber) [reply src/landscape/components/LandscapeListMap.css:22]: what are you trying to override?

File: src/localization/locales/en-US.json (hunk 2 above)
Comment (paulschreiber) [reply src/localization/locales/en-US.json:437]: "Click" (a) doesn't makes sense on mobile devices (b) is not accessible languages. It's also redundant.

File: src/localization/locales/en-US.json (hunk 7 above)
Comment (paulschreiber) [reply src/localization/locales/en-US.json:300]: yeah, that was "for now" — we will need drawing-specific help

**Comment by paulschreiber** [comment]
1. regarding flyover:
image

here you can have three things overlapping and you can't place a point behind Close Shape/Delete Last Point/Cancel

2. It's not clear/obvious when the edit and delete buttons are disabled. they look about the same here:
image

3. the delete button behaviour is confusing. Save and Cancel appear to do the same thing. Can we get rid of the flyover and make clicking the Trash  can do the actual deleting @Hanal-code?
image

**Review by paulschreiber — CHANGES_REQUESTED** [review]
1. regarding flyover:
image

here you can have three things overlapping and you can't place a point behind Close Shape/Delete Last Point/Cancel

2. It's not clear/obvious when the edit and delete buttons are disabled. they look about the same here:
image

3. the delete button behaviour is confusing. Save and Cancel appear to do the same thing. Can we get rid of the flyover and make clicking the Trash  can do the actual deleting @Hanal-code?
image

**Comment by Hanal-code** [comment]
@paulschreiber I was told we can't modify any of those Leaflet draw stuff other than changing the (text)label.

**Comment by josebui** [comment]
@paulschreiber About the Leaflet draw suggestions, those are not possible to customize right now

File: src/common/components/ExternalLink.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import { Link } from '@mui/material';
+
+import { useAnalytics } from 'monitoring/analytics';
+
+const ExternalLink = ({ href, children }) => {
```
Comment (paulschreiber) [inline src/common/components/ExternalLink.js:7]: Add a comment explaining why this component exists and linking to the Plausible GitHub issue.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

**Comment by paulschreiber** [comment]
1. This confirmation message is not helpful and is inconsistent with our other success messages.
>Landscape boundary is complete. Select  to add more detail to your boundary or select Add Landscape to finalize
>your landscape.

2. `Click on a feature to remove` is unclear. What's a feature?

3. `Use  to change the map below` is unclear. We should explain how the map will change (i.e. change the tiles from street to satellite view)

4. The flyout text still blocks the path.

5. Deleting still takes too many clicks.

6. It's not clear disabled buttons are disabled. Effect is too subtle.

**Review by paulschreiber — CHANGES_REQUESTED** [review]
1. This confirmation message is not helpful and is inconsistent with our other success messages.
>Landscape boundary is complete. Select  to add more detail to your boundary or select Add Landscape to finalize
>your landscape.

2. `Click on a feature to remove` is unclear. What's a feature?

3. `Use  to change the map below` is unclear. We should explain how the map will change (i.e. change the tiles from street to satellite view)

4. The flyout text still blocks the path.

5. Deleting still takes too many clicks.

6. It's not clear disabled buttons are disabled. Effect is too subtle.

File: src/localization/locales/en-US.json [hunk 12]
```diff
@@ -395,7 +403,83 @@
...
+            },
+            "edit": {
+                "toolbar": {
+                    "actions": {
+                        "save": {
+                            "title": "Save changes",
+                            "text": "Save"
+                        },
+                        "cancel": {
+                            "title": "Cancel editing, discards all changes",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:453]:
```suggestion
                            "title": "Cancel editing, discarding all changes",
```

File: src/localization/locales/en-US.json [hunk 13]
```diff
@@ -395,7 +403,83 @@
...
+                        "editDisabled": "No polygon to edit",
+                        "remove": "Delete polygon",
+                        "removeDisabled": "No polygon to delete"
+                    }
+                },
+                "handlers": {
+                    "edit": {
+                        "tooltip": {
+                            "text": "Drag points to adjust shape.",
+                            "subtext": "Click the point to remove it."
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:472]:
```suggestion
                            "subtext": "Click a point to remove it."
```

**Review by paulschreiber — CHANGES_REQUESTED** [review]

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**